test/extended/operators/certs.go

```diff
@@ -49,6 +57,123 @@ var _ = g.Describe("[sig-arch][Late]", func() {
+				continue
+			}
+
+			_, err := certgraphutils.LocateCertificateAuthorityBundle(currLocation, expectedPKIContent.CertificateAuthorityBundles)
+			if err != nil {
+				newTLSRegistry.CertificateAuthorityBundles = append(newTLSRegistry.CertificateAuthorityBundles, currCABundle)
+			}
+		}
+
+		if len(newTLSRegistry.CertKeyPairs) > 0 || len(newTLSRegistry.CertificateAuthorityBundles) > 0 {
+			registryString, err := json.MarshalIndent(newTLSRegistry, "", "  ")
+			if err != nil {
+				//g.Fail("Failed to marshal registry %#v: %v", newTLSRegistry, err)
+				testresult.Flakef("Failed to marshal registry %#v: %v", newTLSRegistry, err)
+			}
+			// TODO: uncomment when test no longer fails and enhancement is merged
+			//g.Fail(fmt.Sprintf("Unregistered TLS certificates:\n%s", registryString))
+			testresult.Flakef(fmt.Sprintf("Unregistered TLS certificates found:\n%s\nSee tls/ownership/README.md in origin repo", registryString))
+		}
+	})
+
+	g.It("all registered tls artifacts must have expected metadata", func() {
+
+		ctx := context.Background()
+		kubeClient := oc.AdminKubeClient()
+		if ok, _ := exutil.IsMicroShiftCluster(kubeClient); ok {
+			g.Skip("microshift does not auto-collect TLS.")
+		}
+
+		nodes := map[string]int{}
+		controlPlaneLabel := labels.SelectorFromSet(map[string]string{"node-role.kubernetes.io/control-plane": ""})
+		nodeList, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{LabelSelector: controlPlaneLabel.String()})
+		o.Expect(err).NotTo(o.HaveOccurred())
+		for i, node := range nodeList.Items {
+			nodes[node.Name] = i
+		}
+
+		actualPKIContent, err := certgraphanalysis.GatherCertsFromPlatformNamespaces(ctx, kubeClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		expectedPKIContent, err := certs.GetPKIInfoFromEmbeddedOwnership(ownership.PKIOwnership)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		messages := []string{}
+
+		for _, currCertKeyPair := range actualPKIContent.InClusterResourceData.CertKeyPairs {
+			currLocation := currCertKeyPair.SecretLocation
+			expectedSecret, err := certgraphutils.LocateCertKeyPair(currLocation, expectedPKIContent.CertKeyPairs)
+			if err != nil {
+				continue
+			}
+			errorMessage := o.Equal(expectedSecret.CertKeyInfo).FailureMessage(currCertKeyPair.CertKeyInfo)
+			if len(errorMessage) > 0 {
+				messages = append(messages, fmt.Sprintf("--namespace=%s, secret/%s:\n%v\n", currLocation.Namespace, currLocation.Name, errorMessage))
+			}
+		}
+
+		for _, currCABundle := range actualPKIContent.InClusterResourceData.CertificateAuthorityBundles {
+			currLocation := currCABundle.ConfigMapLocation
+			expectedCABundle, err := certgraphutils.LocateCertificateAuthorityBundle(currLocation, expectedPKIContent.CertificateAuthorityBundles)
+			if err != nil {
+				continue
+			}
+			errorMessage := o.Equal(expectedCABundle.CABundleInfo).FailureMessage(currCABundle.CABundleInfo)
```

Comment:
This looks overly strict.  It would fail on changes to the description, right?

Separate the checking so there is one block for checking owners and another block for checking descriptions, so that someone could copy/paste to have a new section for checking "regenerates on expired certificate"

Reply:
TODO: validate each property separately, maintain separate violation files, mention specific reason in markdown file

Reply:
> TODO: validate each property separately, maintain separate violation files, mention specific reason in markdown file

We already have enough to do that if you scope your PR to Owners only.